packages/lib/modules/lbp/steps/SaleStructureStep.tsx

```diff
@@ -376,6 +342,111 @@ function LaunchTokenAddressInput({
   )
 }
 
+function SaleStartInput({
+  control,
+  errors,
+  value,
+  triggerValidation,
+}: {
+  control: Control<SaleStructureForm>
+  errors: FieldErrors<SaleStructureForm>
+  value: string
+  triggerValidation: UseFormTrigger<SaleStructureForm>
+}) {
+  useEffect(() => {
+    if (value) triggerValidation('startDateTime')
+  }, [value, triggerValidation])
+
+  useInterval(() => {
+    if (value) triggerValidation('startDateTime')
+  }, 5000)
+
+  return (
+    <>
+      <DateTimeInput
+        control={control}
+        errors={errors}
+        label="Start date and time"
+        min={format(addHours(new Date(), 1), "yyyy-MM-dd'T'HH:mm:00")}
+        name="startDateTime"
+        validate={isSaleStartValid}
+      />
+      {saleStartsSoon(value) && !errors['startDateTime'] && (
+        <Text color="font.warning" fontSize="sm">
+          This sale starts soon. Make sure to seed liquidity before this time or the LBP will fail
+          to launch.
+        </Text>
+      )}
+    </>
+  )
+}
+
+function isSaleStartValid(value: string | number) {
+  if (typeof value !== 'string') return 'Start time must be type string'
+
+  if (isBefore(parseISO(value), now())) {
+    return `This start date is already in the past.
+              Set it at least 1 hour ahead to ensure the pool is seeded
+              before the sale begins.`
+  }
+
+  if (!isAfter(parseISO(value), addHours(now(), 1))) {
+    return 'Set the start time at least 1 hour ahead to ensure the pool is seeded before the sale begins.'
+  }
+
+  return true
+}
+
+function saleStartsSoon(saleStart: string) {
+  return (
+    saleStart && isSaleStartValid(saleStart) && isBefore(parseISO(saleStart), addDays(now(), 1))
+  )
+}
+
+function SaleEndInput({
+  control,
+  errors,
+  value,
+  saleStart,
+}: {
+  control: Control<SaleStructureForm>
+  errors: FieldErrors<SaleStructureForm>
+  value: string
+  saleStart: string
+}) {
+  const validateSaleEnd = (value: string | number) => {
+    if (typeof value !== 'string') return 'End time must be type string'
+    if (!isAfter(parseISO(value), addDays(parseISO(saleStart), 1))) {
+      return 'End time must be at least 24 hours after start time'
+    }
+    return true
+  }
```

Comment:
saw you moved similar function `isSaleStartValid` to outside the component context. that is generally best practice because better performance ( but also this case doesnt really matter right? )